src/main/typescript/data/actor/character/magic/source.ts

```diff
@@ -17,6 +22,10 @@ export const MAGIC_JSON_SCHEMA: JSONSchemaType<MagicSource> = {
       description: "The selected Thaumaturgy SPECIAL of the character",
       type: "string",
       enum: ThaumaturgySpecials
+    },
+    magicSpecials: {
+      type: "object",
+      additionalProperties: true
```

Comment:
This can be done better. We might not know the keys, but we definitely know the schema of the values. You can pass entire schemas to `additionalProperties` instead of just `true` and `false`.